crates/sui-core/src/authority_aggregator.rs

```diff
@@ -204,34 +224,85 @@ impl EffectsStakeMap {
         entry.stake += weight;
         entry.signatures.push(sig);
 
-        if entry.stake >= committee.quorum_threshold() {
-            self.effects_cert = CertifiedTransactionEffects::new(
-                entry.effects.clone(),
-                entry.signatures.clone(),
-                committee,
-            )
-            .tap_err(|err| {
-                error!(
-                    "A quorum of effects are available but failed to form a certificate: {:?}",
-                    err
-                );
-            })
-            .ok();
-            self.effects_cert.is_some()
-        } else {
-            false
+        if entry.stake < committee.quorum_threshold() {
+            return Ok(false);
         }
+        let cte = CertifiedTransactionEffects::new(
+            entry.effects.clone(),
+            entry.signatures.clone(),
+            committee,
+        )
+        .tap_err(|err| {
+            error!(
+                "A quorum of effects are available but failed to form a certificate: {:?}",
+                err
+            );
+        })
+        .map_err(|e| EffectsCertError {
+            error: e,
+            effect_digest: entry.effects.digest(),
+            authorities: entry.signatures.iter().map(|s| s.authority).collect(),
+            total_stake: entry.stake,
+        })?
```

Comment:
it's actually very tempting for me to just unwrap here and in line 253, because I don't think they are reachable:
1. individual response is verified in `SafeClient` before reaching here, so all signatures are valid. 
2. in terms of epoch change, we only look at an effects when the epoch matches the current committee.

@joyqvq  do you know if this could error on other sides?